types/src/account_address.rs

```diff
@@ -216,3 +216,21 @@ impl TryFrom<AccountAddress> for Bech32 {
         bech32::Bech32::new(LIBRA_NETWORK_ID_SHORT.into(), base32_hash).map_err(Into::into)
     }
 }
+
+pub fn deserialize_as_str<'de, D>(deserializer: D) -> std::result::Result<AccountAddress, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let account_address_string: String = Deserialize::deserialize(deserializer)?;
+    AccountAddress::try_from(account_address_string).map_err(<D::Error as serde::de::Error>::custom)
+}
+
+pub fn serialize_as_str<S>(
+    account_address: &AccountAddress,
+    serializer: S,
+) -> std::result::Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_str(&account_address.to_string())
+}
```

Comment:
oh you did add these in this commit...maybe we should take a closer look then...Basically we can have the impl of Ser/De handle both of these cases transparently using the `Deserializer::is_human_readable()` function to figure out if we should de some bytes or a str. either way not a blocker at all

Reply:
Oh, hmm... https://docs.serde.rs/serde/trait.Serializer.html#method.is_human_readable

I hadn't seen that before... let me improve that commit. I wasn't happy with what I had done :)

Reply:
this has been replaced by implementing the serialize / deserialize trait.